x-pack/plugins/upgrade_assistant/__jest__/client_integration/kibana_deprecations/deprecations_list.test.ts

```diff
@@ -0,0 +1,138 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { deprecationsServiceMock } from 'src/core/public/mocks';
+
+import { setupEnvironment } from '../helpers';
+import { KibanaTestBed, setupKibanaPage } from './kibana_deprecations.helpers';
+import { kibanaDeprecationsMockResponse } from './mocked_responses';
+
+const criticalDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.level === 'critical'
+);
+const warningDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.level === 'warning'
+);
+const configDeprecations = kibanaDeprecationsMockResponse.filter(
+  (deprecation) => deprecation.deprecationType === 'config'
+);
+
+describe('Kibana deprecations table', () => {
+  let testBed: KibanaTestBed;
+  const { server } = setupEnvironment();
+  const deprecationService = deprecationsServiceMock.createStartContract();
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  beforeEach(async () => {
+    await act(async () => {
+      deprecationService.getAllDeprecations = jest
+        .fn()
+        .mockReturnValue(kibanaDeprecationsMockResponse);
+
+      testBed = await setupKibanaPage({
+        services: {
+          core: {
+            deprecations: deprecationService,
+          },
+        },
+      });
+    });
+
+    testBed.component.update();
+  });
+
+  test('renders deprecations', () => {
+    const { exists, table } = testBed;
+
+    expect(exists('kibanaDeprecations')).toBe(true);
+
+    const { tableCellsValues } = table.getMetaData('kibanaDeprecationsTable');
+
+    expect(tableCellsValues.length).toEqual(kibanaDeprecationsMockResponse.length);
+  });
+
+  it('refreshes deprecation data', async () => {
+    const { actions } = testBed;
+
+    await actions.table.clickRefreshButton();
+
+    expect(deprecationService.getAllDeprecations).toHaveBeenCalledTimes(2);
+  });
+
+  it('shows critical and warning deprecations count', () => {
+    const { find } = testBed;
+
+    expect(find('criticalDeprecationsCount').text()).toContain(criticalDeprecations.length);
+    expect(find('warningDeprecationsCount').text()).toContain(warningDeprecations.length);
+  });
+
+  describe('Search bar', () => {
+    it('filters by "critical" status', async () => {
+      const { actions, table } = testBed;
+
+      await actions.searchBar.clickCriticalFilterButton();
+
+      const { rows: criticalRows } = table.getMetaData('kibanaDeprecationsTable');
+
+      expect(criticalRows.length).toEqual(criticalDeprecations.length);
+
+      await actions.searchBar.clickCriticalFilterButton();
+
+      const { rows: allRows } = table.getMetaData('kibanaDeprecationsTable');
+
+      expect(allRows.length).toEqual(kibanaDeprecationsMockResponse.length);
```

Comment:
Nit: I'd find this easier to read if we grouped these lines into 2 groups with comments to describe them.

```js
      // Show only critical deprecations
      await actions.searchBar.clickCriticalFilterButton();
      const { rows: criticalRows } = table.getMetaData('kibanaDeprecationsTable');
      expect(criticalRows.length).toEqual(criticalDeprecations.length);

      // Show all deprecations
      await actions.searchBar.clickCriticalFilterButton();
      const { rows: allRows } = table.getMetaData('kibanaDeprecationsTable');
      expect(allRows.length).toEqual(kibanaDeprecationsMockResponse.length);
```